prog/weaver/main.go

```diff
@@ -293,7 +293,11 @@ func main() {
 		}
 		router.HandleHTTP(muxRouter)
 		HandleHTTP(muxRouter, version, router, allocator, defaultSubnet, ns, dnsserver)
-		http.Handle("/", muxRouter)
+		var handler http.Handler = muxRouter
+		if logLevel == "debug" {
+			handler = common.LogHandler(muxRouter)
+		}
+		http.Handle("/", handler)
```

Comment:
Is there any harm in *always* going via the logging handler? We really don't need to be concerned about efficiency here.

That way all of the above collapses to

    http.Handle("/", common.LoggingHTTPHandler(muxRouter))